abel/tools/circularize.py

```diff
@@ -0,0 +1,236 @@
+        radius slice profile.
+
+    smooth: float
+        smoothing for spline interpolation of the determined radial scaling
+        factor vs angle
+
+    nslices: int
+        divide the image into nslices. Ideally, this should be a divisor of
+        the image column width.
+
+    ref_angle: None or float
+        reference angle for which radial coordinate is unchanged.
+        None uses numpy.mean(radial scale factors)
+
+    inverse: bool
+        inverse Abel transform the *polar* image, to remove the background
+        intensity. This helps follow a given peak intensityr, for all angles,
+        when the anisotropy parameter is large.
+
+    return_correction: bool
+        additional outputs, see below
+
+    Returns
+    -------
+    IMcirc : numpy 2D array, same size as input
+        Circularized imput image
+
+    (if return_correction is True)
+    slice_angles: numpy 1D array
+        Mid-point angle (radians) of image slice
+
+    radial_corrections: numpy 1D array
+        radial correction scale factors, at each angle
+
+    spline_function: numpy func()
+        spline function used to evaluate the radial correction at each angle
+
+    """
+
+    if zoom > 1:
+        # zoom may improve the low radii angular resolution
+        IM = ndimage.zoom(IM, zoom)
+
+    if center is not None:
+        # convenience function for the case image is not centered
+        IM = abel.tools.center.center_image(IM, center=center)
+
+    # map image into polar coordinates - much easier to slice
+    # cartesian (Y, X) -> polar (Radius, Theta)
+    polarIM, radial_coord, angle_coord =\
+                           abel.tools.polar.reproject_image_into_polar(IM)
+ 
+    if inverse:
+        # pseudo inverse Abel transform of the polar image, removes background
+        # to enhance transition peaks
+        polarIM = abel.dasch.two_point_transform(polarIM.T).T
+
+    # limit radial range of polar image, if selected
+    radial = radial_coord[:, 0]
+    if radial_range is not None:
+        subr = np.logical_and(radial > radial_range[0]*zoom,
+                              radial < radial_range[1]*zoom)
+        polarIM = polarIM[subr]
+        radial = radial[subr]
+
+    # split image into n-slices
+    slices = np.array_split(polarIM, nslices, axis=1)
+    slice_angles_array = np.array_split(angle_coord[0], nslices)
+    # Fix me! - np.hsplit does not cope with uneven split
+    #           there must be a better way to determine mean slice angle?
+    slice_angles = []
+    for ang in slice_angles_array:
+        slice_angles.append(np.mean(ang))
+    slice_angles = np.array(slice_angles)
+
+    # evaluate radial scaling factor for each slice
+    radcorr = correction(slice_angles, slices, radial, method=method)
+
+    # spline radial scaling vs angle
+    radcorrspl = UnivariateSpline(slice_angles, radcorr, s=smooth, ext=3)
+
+    # apply the correction
+    IMcirc = circularize(IM, radcorrspl, ref_angle=ref_angle)
+
+    if zoom > 1:
```

Comment:
Even if we are going to use the "zoom" concept (which I think can be eliminated through `dr` as discussed above), then we could simply apply the correction to the un-zoomed image and eliminate this line here.